lib/timeutils/strptime-tz.c

```diff
@@ -208,7 +210,7 @@ first_wday_of(int yr)
  * their struct tm. This is a slightly modified NetBSD strptime() with
  * some modifications and explicit zone related parameters. */
 char *
-strptime_with_tz(const char *buf, const char *fmt, struct tm *tm, long *tm_gmtoff, const char **tm_zone)
+strptime_with_tz(const char *buf, const char *fmt, struct tm *tm, long *tm_gmtoff, const char **tm_zone, int *tm_usec)
 {
```

Comment:
I increasingly have the feeling that this should be parsing into a WallClockTime instance, instead of littering its argument list with more and more values.

The only reason I didn't do this previously was that I tried to keep it as close to the original NetBSD implementation as possible. But we have forked this function anyway and I don't expect it to sync with upstream.

If that's the case we could possibly remove some code from it as well. This is not a blocking comment.

Reply:
Yeah, I share this view: I first hacked in the area that seems to be the right one, and was happy to discover that `strptime_with_tz()` was called in a single place :smile: I did not renamed the function (`strptime_with_tz_and_usec_and_probably_other_stuff_in_the_future()`), but I makes sense to move the code at the `wall_clock_time_strptime()` level.

Maybe we can solve all the minor glitches and feedback before adding a commit that just moves the code around without changing it?

Reply:
ok, makes sense.